### PR TITLE
adds tutorial

### PR DESCRIPTION
**Pull request by MsUzoAgu**
@Danny-Sullivan : as discussed the tutorial branch is finished.

@SophieDeBenedetto and @deniznida: take a look a look please?

### CODE REVIEW

File: TUTORIAL.md [hunk 1]
```diff
@@ -0,0 +1,897 @@
...
+  - get paid/receive payments
+  - take a bath
+  - call a friend
+  - start a conversation
+  - state if they are happy and/or clean
+
+## Objectives
+1. Gain proficiency instantiating a class
+2. Gain ability to discern when to implement `attr_accessor`, `attr_reader`,  and `attr_writer`
+3. Gain abiity to discern when to define your own `attr_reader`(getter) and `attr_writer`(setter) methods
```
Comment (deniznida) [inline TUTORIAL.md]: there is a typo in #3 (Gain ability)

File: TUTORIAL.md [hunk 2]
```diff
@@ -0,0 +1,897 @@
...
+     NameError:
+       uninitialized constant Person
+     # ./spec/person_spec.rb:3:in `block (2 levels) in <top (required)>'
+     # ./spec/person_spec.rb:6:in `block (2 levels) in <top (required)>'
+```
+
+Our 1st hint is the NameError `uninitialized contant Person`.  This error points to the fact that we do not have a person class.
+
+Open up the person.rb file in lib directory and write the code that will get this test to pass. And then run rspec with the fail fast option, `rspec --f-f`, to get our next hint.
+
```
Comment (deniznida) [inline TUTORIAL.md]: - What is the code that will make the test pass?
- Can you please update the code snippet to `rspec --fail-fast`?

File: TUTORIAL.md [hunk 3]
```diff
@@ -0,0 +1,897 @@
...
+Failures:
+  1) Person instantiation ::new a new person is instantiated with a name
+     Failure/Error: let(:stella) { Person.new("Stella") }
+     ArgumentError:
+       wrong number of arguments (1 for 0)
+     # ./spec/person_spec.rb:3:in `initialize'
+     # ./spec/person_spec.rb:3:in `new'
+     # ./spec/person_spec.rb:3:in `block (2 levels) in <top (required)>'
+     # ./spec/person_spec.rb:6:in `block (2 levels) in <top (required)>'
+
```
Comment (deniznida) [inline TUTORIAL.md]: please take out the lines 55-58

File: TUTORIAL.md [hunk 4]
```diff
@@ -0,0 +1,897 @@
...
+Failures:
+
+  1) Person instantiation #initialize #name a new person knows its name once he/she has been initialized
+     Failure/Error: expect(person.name).to eq(name)
+     NoMethodError:
+       undefined method `name' for #<Person:0x007f992d0ac508 @name="Stella">
+     # ./spec/person_spec.rb:19:in `block (6 levels) in <top (required)>'
+     # ./spec/person_spec.rb:18:in `each'
+     # ./spec/person_spec.rb:18:in `block (5 levels) in <top (required)>'
+
```
Comment (deniznida) [inline TUTORIAL.md]: we can take out lines 97-99

File: TUTORIAL.md [hunk 5]
```diff
@@ -0,0 +1,897 @@
...
+Failures:
+  1) Person instantiation #initialize #bank_account a new person instance is initialized with a bank_account balance of $25
+     Failure/Error: people.each { |person| expect(person.bank_account).to eq(25) }
+     NoMethodError:
+       undefined method `bank_account' for #<Person:0x007fd0714cd3f0 @name="Stella">
+     # ./spec/person_spec.rb:32:in `block (6 levels) in <top (required)>'
+     # ./spec/person_spec.rb:32:in `each'
+     # ./spec/person_spec.rb:32:in `block (5 levels) in <top (required)>'
+
```
Comment (deniznida) [inline TUTORIAL.md]: please take out lines 136-138

File: TUTORIAL.md [hunk 6]
```diff
@@ -0,0 +1,897 @@
...
+Failures:
+
+  1) Person instantiation #initialize #bank_account a person instance can change his/her bank_account amount
+     Failure/Error: expect(person.bank_account += 1).to eq (original_amount + 1)
+     NoMethodError:
+       undefined method `bank_account=' for #<Person:0x007fa6cba6d3c0 @name="Stella", @bank_account=25>
+     # ./spec/person_spec.rb:38:in `block (6 levels) in <top (required)>'
+     # ./spec/person_spec.rb:36:in `each'
+     # ./spec/person_spec.rb:36:in `block (5 levels) in <top (required)>'
+
```
Comment (deniznida) [inline TUTORIAL.md]: again please take out lines 176-178

File: TUTORIAL.md [hunk 7]
```diff
@@ -0,0 +1,897 @@
...
+Failures:
+  1) Person instantiation #initialize #happiness  a new person instance is initialized with a happiness index of 8
+     Failure/Error: people.each { |person| expect(person.happiness).to eq(8)}
+     NoMethodError:
+       undefined method `happiness' for #<Person:0x007fd2ac3ab228 @name="Stella", @bank_account=25>
+     # ./spec/person_spec.rb:45:in `block (6 levels) in <top (required)>'
+     # ./spec/person_spec.rb:45:in `each'
+     # ./spec/person_spec.rb:45:in `block (5 levels) in <top (required)>'
+
```
Comment (deniznida) [inline TUTORIAL.md]: we don't need lines 214-216

File: TUTORIAL.md [hunk 8]
```diff
@@ -0,0 +1,897 @@
...
+Failures:
+
+  1) Person instantiation #initialize #happiness a person instance can change his/her happiness index
+     Failure/Error: person.happiness += 1
+     NoMethodError:
+       undefined method `happiness=' for #<Person:0x007fb5db9f1b70>
+     # ./spec/person_spec.rb:51:in `block (6 levels) in <top (required)>'
+     # ./spec/person_spec.rb:49:in `each'
+     # ./spec/person_spec.rb:49:in `block (5 levels) in <top (required)>'
+```
```
Comment (deniznida) [inline TUTORIAL.md]: don't need lines 253-255

File: TUTORIAL.md [hunk 9]
```diff
@@ -0,0 +1,897 @@
...
+    @bank_account = 25
+    @happiness = 8
+  end
+end
+```
+When we run rspec again we should see this error message:
+
+```ruby
+Person
+  (passing code ommitted for brevity)
```
Comment (deniznida) [inline TUTORIAL.md]: typo in omitted

File: TUTORIAL.md [hunk 10]
```diff
@@ -0,0 +1,897 @@
...
+
+       expected: 10
+            got: 108
+
+       (compared using ==)
+     # ./spec/person_spec.rb:60:in `block (6 levels) in <top (required)>'
+     # ./spec/person_spec.rb:57:in `each'
+     # ./spec/person_spec.rb:57:in `block (5 levels) in <top (required)>'
+```
+
```
Comment (deniznida) [inline TUTORIAL.md]: don't need lines 292-294

File: TUTORIAL.md [hunk 11]
```diff
@@ -0,0 +1,897 @@
...
+Hmmm ... this is interesting.  It appears that we need to define our setter/attr_writter for happiness in such a way as that its index never greater than 10.
+
```
Comment (deniznida) [inline TUTORIAL.md]: typo in attr_writer

File: TUTORIAL.md [hunk 12]
```diff
@@ -0,0 +1,897 @@
...
+See if you can get the above test to pass and then compare your answer to the one below. Do not fret if your code does not explictly matche the solution - the two most important things are that:
+
```
Comment (deniznida) [inline TUTORIAL.md]: typo -> match the solution

File: TUTORIAL.md [hunk 13]
```diff
@@ -0,0 +1,897 @@
...
+  end
+end
+```
+
+The next error we see deals with setting a limit on how unhappy a person can be:
+
+```ruby
+Person
+  (passing code ommitted for brevity)
+  #happiness
```
Comment (deniznida) [inline TUTORIAL.md]: typo -> omitted

File: TUTORIAL.md [hunk 14]
```diff
@@ -0,0 +1,897 @@
...
+     Failure/Error: expect(person.happiness).to eq(0)
+
+       expected: 0
+            got: -92
+
+       (compared using ==)
+     # ./spec/person_spec.rb:68:in `block (6 levels) in <top (required)>'
+     # ./spec/person_spec.rb:65:in `each'
+     # ./spec/person_spec.rb:65:in `block (5 levels) in <top (required)>'
+
```
Comment (deniznida) [inline TUTORIAL.md]: don't need lines 340-342

File: TUTORIAL.md [hunk 15]
```diff
@@ -0,0 +1,897 @@
...
+```
+
+We get the above test to pass by simply adding a line that handles this specific situation to our setter/attr_writter for happiness:
+
```
Comment (deniznida) [inline TUTORIAL.md]: typo -> attr_writer

File: TUTORIAL.md [hunk 16]
```diff
@@ -0,0 +1,897 @@
...
+  end
+end
+```
+
+## RSpec Test 5: `#hygiene`
+
+```ruby
+Person
+  (passing code ommitted for brevity)
+  #hygiene
```
Comment (deniznida) [inline TUTORIAL.md]: typo -> omitted

File: TUTORIAL.md [hunk 17]
```diff
@@ -0,0 +1,897 @@
...
+
+Failures:
+  1) Person instantiation #initialize #hygiene a person instance is initialized with a hygiene index of 8
+     Failure/Error: people.each { |person| expect(person.hygiene).to eq(8) }
+     NoMethodError:
+       undefined method `hygiene' for #<Person:0x007f94c95dc3b0>
+     # ./spec/person_spec.rb:75:in `block (6 levels) in <top (required)>'
+     # ./spec/person_spec.rb:75:in `each'
+     # ./spec/person_spec.rb:75:in `block (5 levels) in <top (required)>'
+```
```
Comment (deniznida) [inline TUTORIAL.md]: don't need lines 381-383

File: TUTORIAL.md [hunk 18]
```diff
@@ -0,0 +1,897 @@
...
+
+Reading through line 73 - 99 in `spec/person_spec.rb` and comparing it to lines 43 - 71 we should notice the similarities between the tests for `#hygiene` and `#happiness`.
+
+Given that we have just worked on `#happiness` see if you can get all the tests for `#hygiene` to pass.  Once you have given the tests a try (and hopefully gotten your code to pass the test) compare your solution to the solution below.
+
+Again, do not fret if your code does not explictly matche the solution - the two most important things are that:
```
Comment (deniznida) [inline TUTORIAL.md]: typo -> match

File: TUTORIAL.md [hunk 19]
```diff
@@ -0,0 +1,897 @@
...
+     NoMethodError:
+       undefined method `happy?' for #<Person:0x007fd14504c400>
+     # ./spec/person_spec.rb:114:in `block (4 levels) in <top (required)>'
+```
+
+NoMethodError means we need to define `#happy?`. Also according to line 112 in `spec/person_spec.rb` `#happy`  needs to `"returns true if happiness is greater than seven, else false"`. We can get this test to pass this way:
+
+```ruby
+class person
+  (previous code ommitted for brevity)
```
Comment (deniznida) [inline TUTORIAL.md]: typo -> omitted

File: TUTORIAL.md [hunk 20]
```diff
@@ -0,0 +1,897 @@
...
+end
+```
+
+The above code is very explicit (and verbose). Can you think of a one-line code that would also achieve the same result? Refactor your code if you.
+
+## RSpec Test 7: `#clean?`
+
+```ruby
+Person
+  (passing code ommitted for brevity)
```
Comment (deniznida) [inline TUTORIAL.md]: typo -> omitted

File: TUTORIAL.md [hunk 21]
```diff
@@ -0,0 +1,897 @@
...
+       undefined method `clean?' for #<Person:0x007fa0848f15c0>
+     # ./spec/person_spec.rb:123:in `block (4 levels) in <top (required)>'
+```
+
+Notice the similarity between the tests for `#clean?` and `#happy?`(compare lines 111-118 and lines 120 - 127)? Therefore our method definition for `#clean` looks like this:
+
+```ruby
+class Person
+
+(previous code ommitted for brevity)
```
Comment (deniznida) [inline TUTORIAL.md]: typo -> omitted

File: TUTORIAL.md [hunk 22]
```diff
@@ -0,0 +1,897 @@
...
+
+> Point to consider:
+> - do we need the `self` in front of hygiene? What happens if we remove it? See if you can explain in plain english why our code still passes if we remove `self`
+
+
+## RSpec Test 8: `#get_paid`
+
+```ruby
+Person
+  (passing code omitted for brevity)
```
Comment (deniznida) [inline TUTORIAL.md]: typo -> omitted

File: TUTORIAL.md [hunk 23]
```diff
@@ -0,0 +1,897 @@
...
+
+Failures:
+  1) Person non-attribute instance methods #get_paid accepts an argument of salary
+     Failure/Error: expect { penelope.get_paid(100) }.to_not raise_error
+       expected no Exception, got #<NoMethodError: undefined method `get_paid' for #<Person:0x007fd89211a7c0>> with backtrace:
+         # ./spec/person_spec.rb:131:in `block (5 levels) in <top (required)>'
+         # ./spec/person_spec.rb:131:in `block (4 levels) in <top (required)>'
+     # ./spec/person_spec.rb:131:in `block (4 levels) in <top (required)>'
+```
+
```
Comment (deniznida) [inline TUTORIAL.md]: don't need lines 504-506

File: TUTORIAL.md [hunk 24]
```diff
@@ -0,0 +1,897 @@
...
+Read lines 129 - 143 in `spec/person_spec.rb` to get a better understanding of the requirements for the `#get_paid` method and then write out the code that gets the test to pass.
+
+```ruby
+class Person
+(previous code ommitted for brevity)
+
```
Comment (deniznida) [inline TUTORIAL.md]: typo -> omitted

File: TUTORIAL.md [hunk 25]
```diff
@@ -0,0 +1,897 @@
...
+Failures:
+  1) Person non-attribute instance methods #call_friend accepts one argument, an instance of the Person class
+     Failure/Error: expect { penelope.call_friend(felix) }.to_not raise_error
+       expected no Exception, got #<NoMethodError: undefined method `call_friend' for #<Person:0x007ff992121928>> with backtrace:
+         # ./spec/person_spec.rb:220:in `block (5 levels) in <top (required)>'
+         # ./spec/person_spec.rb:220:in `block (4 levels) in <top (required)>'
+     # ./spec/person_spec.rb:220:in `block (4 levels) in <top (required)>'
+
+```
```
Comment (deniznida) [inline TUTORIAL.md]: don't need lines 712-714

**Comment by deniznida** [comment]
this needs to be a PR to solution

File: TUTORIAL.md [hunk 26]
```diff
@@ -0,0 +1,897 @@
...
+     Failure/Error: let(:stella) { Person.new("Stella") }
+     NameError:
+       uninitialized constant Person
+     # ./spec/person_spec.rb:3:in `block (2 levels) in <top (required)>'
+     # ./spec/person_spec.rb:6:in `block (2 levels) in <top (required)>'
+```
+
+Our 1st hint is the NameError `uninitialized contant Person`.  This error points to the fact that we do not have a person class.
+
+Open up the person.rb file in lib directory and write the code that will get this test to pass. And then run rspec with the fail fast option, `rspec --f-f`, to get our next hint.
```
Comment (MsUzoAgu) [inline TUTORIAL.md]: @deniznida: don't understand the question. They should write code that gets that particular test to pass. Is it the wording that is the problem?

Comment (deniznida) [inline TUTORIAL.md]: No it is not wording, you wrote "write the code that will get this test to pass" but you did not add an example of that code. 
And I believe the `rspec --f-f` is not default to everyones computer thats why I said if you could please update it to `rspec --fail-fast`

Comment (MsUzoAgu) [inline TUTORIAL.md]: @deniznida: oh! that's because the idea is to get them to attempt writing an answer that gets the first test to pass. If they can't (and thus they get an error) then we guide them through resolving the error.

Comment (MsUzoAgu) [inline TUTORIAL.md]: @deniznida : Got it! Thanks :)

**Comment by MsUzoAgu** [comment]
@deniznida: can you take another look please? Thanks lady :)

**Comment by deniznida** [comment]
@MsUzoAgu The TUTORIAL.md file does not need the objectives and intro other than that looks good. Thank you!!
And please open new PR into solution
